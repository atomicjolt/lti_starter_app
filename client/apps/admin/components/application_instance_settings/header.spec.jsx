import React from 'react';
import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';

import Header from './header';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

const settings =  {
  sign_out_url: 'https://www.example.com',
};

const store = mockStore({
  applicationInstances: {
    newApplicationInstance: {
    }
  },
  settings,
});

describe('Header', () => {

  const applicationInstance = {
    site: {},
  };
  const application = {};

  it('renders the Header component', () => {
    const result = TestRenderer.create(
      <Provider store={store}>
        <Header applicationInstance={applicationInstance} application={application}/>
      </Provider>
    );
    expect(result).toMatchSnapshot();
  });
});
