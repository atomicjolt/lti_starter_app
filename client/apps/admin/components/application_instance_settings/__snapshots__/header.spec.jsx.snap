// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the Header component 1`] = `
<div
  className="aj-header"
>
  <button
    className="back-btn"
    type="button"
  >
    <i
      className="fas fa-chevron-left aj-icon"
    />
    Â Instances
  </button>
  <div
    className="header-row space-between"
  >
    <div
      className="title"
    />
    <div
      className="flex-center"
    >
      <button
        className="aj-btn"
        onClick={[Function]}
        type="button"
      >
        Save
      </button>
      <div
        className="aj-menu-contain "
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="Instance Settings"
          className="aj-icon-btn"
          onClick={[Function]}
          type="button"
        >
          <i
            aria-hidden="true"
            className="material-icons"
          >
            more_vert
          </i>
        </button>
        <ul
          className="aj-menu"
          role="menu"
        >
          <li>
            <button
              className="c-disable"
              onClick={[Function]}
              type="button"
            >
              <i
                className="material-icons enabled"
              >
                check_circle
              </i>
              Disable
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
              type="button"
            >
              <i
                className="material-icons"
              >
                delete
              </i>
              Delete
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="header-row info"
  >
    <div
      className="info-spacing"
    >
      <div>
        Primary Contact
      </div>
      <a
        href="mailto: undefined"
      />
    </div>
    <div
      className="info-spacing"
    >
      <div>
        Canvas URL
      </div>
      <a />
    </div>
  </div>
</div>
`;
